apps/server/src/modules/school/api/dto/response/system.response.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { SystemProvisioningStrategy } from '@shared/domain/interface/system-provisioning.strategy';
+import { LdapConfig, OauthConfig, OidcConfig } from '../../../domain';
+
+export class SystemResponse {
+	@ApiProperty()
+	id!: string;
+
+	@ApiProperty()
+	type!: string;
+
+	@ApiPropertyOptional()
+	url?: string;
+
+	@ApiPropertyOptional()
+	alias?: string;
+
+	@ApiPropertyOptional()
+	displayName?: string;
+
+	// TODO: How deeply do we want to be explicit about the response types?
+	// E.g. do we want an OauthConfigResponse here with all properties marked as ApiProperty?
+	@ApiPropertyOptional()
+	oauthConfig?: OauthConfig;
+
+	@ApiPropertyOptional()
+	oidcConfig?: OidcConfig;
+
+	@ApiPropertyOptional()
+	ldapConfig?: LdapConfig;
```

Comment:
Ich bin mir ziemlich sicher das dort mehr drin steht als wir bei einem normalen response raus geben wollen/sollen.
searchUserPassword zum Beispiel. Bisher ist es aber so im legacy implementiert?

Zu deiner Frage, theoretisch sind die Dekoratoren erfoderlich. Ansonsten sind die Information in swagger > openApi und damit auch nicht in den / die clients verfügbar. Ist das Teil einer Schnittstelle die von vue / legacy client gecalled wird?

Reply:
Sehr guter Punkt, dass da zu viel zurückgegeben wird!! Ich hatte da nicht ausreichend drüber nachgedacht, die populiert zurückzugeben. Für die Authentication im alten Client sollen sie nämlich nicht populiert sein.

Ich habe es jetzt so geändert, dass nur die SystemIds zurückgegeben werden. Das entspricht ungefähr dem alten Verhalten und im Nuxt-Client sind dadurch wesentlich weniger Änderungen nötig. Nur im alten Client müssen die Systems jetzt für die Admin-Page separat gefetcht werden, aber ich denke, das ist übergangsweise okay.

Außerdem hab ich alle Systems betreffenden Typen aus diesem PR entfernt. Du hattest ja an anderer Stelle angemerkt, dass die mit den bestehenden auseinanderlaufen könnten. Die Systems kann man dann gut in einem separaten PR umziehen.

Reply:
ok